src/assets/all_assets_request.rs

```diff
@@ -107,9 +73,15 @@ impl AllAssetsRequest {
     /// # Returns
     /// The request object
     /// [AllAssetsRequest](struct.AllAssetsRequest.html)
-    pub fn set_asset_code(&mut self, asset_code: &str) -> &mut Self {
-        self.asset_code = Some(asset_code.to_owned());
-        self
+    pub fn set_asset_code(self, asset_code: &str) -> Result<AllAssetsRequest, String> {
```

Comment:
We should add the boundaries to the documentation, so the developers know if they mess up, why their code doesn't compile.

Reply:
Moved to https://github.com/Baseflow/stellar-rust-sdk/issues/4